### PR TITLE
Add check subscription startdate displayed

### PR DESCRIPTION
**Pull request by swadeley**
Hello

A simple check but a lot of setup so I have added it to the end_to_end test.

This is closed loop for:
[Bug 1920860 – The Start Date field is blank for Subscriptions within Content Hosts page in Satellite WebUI](https://bugzilla.redhat.com/show_bug.cgi?id=1920860)

but with workaround for bug:
[Bug 2112320](https://bugzilla.redhat.com/show_bug.cgi?id=2112320) - Hidden column between Quantity and Attached in the table of subscriptions

### CODE REVIEW

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end

**Review by ColeHiggins2 — APPROVED** [review]
Looks good to me

**Review by latran — APPROVED** [review]

File: tests/foreman/ui/test_contenthost.py [hunk 1]
```diff
@@ -172,6 +174,22 @@ def test_positive_end_to_end(session, default_location, module_repos_collection_
...
+                if item["Repository Name"] == custom_product_name
+            )
+            assert startdate in custom_sub['Expires']
+        else:
+            custom_sub = next(
+                item
+                for item in chost['subscriptions']['resources']['assigned']
+                if item["Repository Name"] == custom_product_name
+            )
+            assert startdate in custom_sub['Starts']
```
Comment (jameerpathan111) [inline tests/foreman/ui/test_contenthost.py:192]:
```suggestion
        # Check start date for BZ#1920860 (but handle BZ#2112320 offset-by-one bug)
        custom_product_name = module_repos_collection_with_manifest.custom_product['name']
        custom_sub = next(
                item
                for item in chost['subscriptions']['resources']['assigned']
                if item["Repository Name"] == custom_product_name
            )
        if is_open('BZ:2112320'):
            assert startdate in custom_sub['Expires']
        else:
            assert startdate in custom_sub['Starts']
```

**Review by jameerpathan111 — CHANGES_REQUESTED** [review]

**Review by mshriver — APPROVED** [review]

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end

File: tests/foreman/ui/test_contenthost.py [hunk 2]
```diff
@@ -172,6 +174,17 @@ def test_positive_end_to_end(session, default_location, module_repos_collection_
             for repo_index in range(len(module_repos_collection_with_manifest.repos_info))
         }
         assert actual_repos == expected_repos
+        # Check start date for BZ#1920860 (but handle BZ#2112320 offset-by-one bug)
+        custom_product_name = module_repos_collection_with_manifest.custom_product['name']
+        custom_sub = next(
+                item
+                for item in chost['subscriptions']['resources']['assigned']
+                if item["Repository Name"] == custom_product_name
+            )
```
Comment (mshriver) [inline tests/foreman/ui/test_contenthost.py:183]:
```suggestion
            item
            for item in chost['subscriptions']['resources']['assigned']
            if item["Repository Name"] == custom_product_name
            )
```

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end

**Review by jameerpathan111 — APPROVED** [review]

**Comment by swadeley** [comment]
Thank you.

**Pull request merged**